ipfs-cluster-service/main.go

```diff
@@ -635,5 +610,75 @@ func makeConfigs() (*config.Manager, *ipfscluster.Config, *rest.Config, *ipfshtt
 	cfg.RegisterComponent(config.Monitor, monCfg)
 	cfg.RegisterComponent(config.Informer, diskInfCfg)
 	cfg.RegisterComponent(config.Informer, numpinInfCfg)
-	return cfg, clusterCfg, apiCfg, ipfshttpCfg, consensusCfg, trackerCfg, monCfg, diskInfCfg, numpinInfCfg
+	return cfg, &cfgs{clusterCfg, apiCfg, ipfshttpCfg, consensusCfg, trackerCfg, monCfg, diskInfCfg, numpinInfCfg}
+}
+
+type cfgs struct {
+	clusterCfg   *ipfscluster.Config
+	apiCfg       *rest.Config
+	ipfshttpCfg  *ipfshttp.Config
+	consensusCfg *raft.Config
+	trackerCfg   *maptracker.Config
+	monCfg       *basic.Config
+	diskInfCfg   *disk.Config
+	numpinInfCfg *numpin.Config
+}
+
+func initializeCluster(ctx context.Context, c *cli.Context, cfgs *cfgs) (*ipfscluster.Cluster, error) {
+	host, err := ipfscluster.NewClusterHost(ctx, cfgs.clusterCfg)
+	checkErr("creating libP2P Host", err)
+
+	api, err := rest.NewAPIWithHost(cfgs.apiCfg, host)
+	checkErr("creating REST API component", err)
+
+	proxy, err := ipfshttp.NewConnector(cfgs.ipfshttpCfg)
+	checkErr("creating IPFS Connector component", err)
+
+	state := mapstate.NewMapState()
+
+	err = validateVersion(cfgs.clusterCfg, cfgs.consensusCfg)
+	checkErr("validating version", err)
+
+	tracker := maptracker.NewMapPinTracker(cfgs.trackerCfg, cfgs.clusterCfg.ID)
+	mon, err := basic.NewMonitor(cfgs.monCfg)
+	checkErr("creating Monitor component", err)
+	informer, alloc := setupAllocation(c.GlobalString("alloc"), cfgs.diskInfCfg, cfgs.numpinInfCfg)
+
+	return ipfscluster.NewCluster(
+		host,
+		cfgs.clusterCfg,
+		cfgs.consensusCfg,
+		api,
+		proxy,
+		state,
+		tracker,
+		mon,
+		alloc,
+		informer,
+	)
+}
+
+func handleCtrlC(cluster *ipfscluster.Cluster, ctrlcCount int) {
+	switch ctrlcCount {
+	case 1:
+		go func() {
+			time.Sleep(1 * time.Minute)
```

Comment:
Of course, stupid me forgets to remove the sleep statement before pushing up 🤦‍♂️ . Needed it to test cluster 'hanging'.